app/scripts/controllers/rif/rns/rns-delegate.js

```diff
@@ -0,0 +1,16 @@
+export default class RnsDelegate {
+  constructor (props) {
+    this.web3 = props.web3;
+    this.preferencesController = props.preferencesController;
+    this.networkController = props.networkController;
+    this.rifConfig = props.rifConfig;
+    this.rnsContractInstance = props.rnsContractInstance;
+    this.selectedAccount = props.selectedAccount;
```

Comment:
What is the idea of having the `selectedAccount `of the plugin as a `RnsDelegate `member?

If the idea is "I want to have the selected account to register/resolver domains with that account" the RnsDelegate must not know about the logic of the plugin. Its a service layer that can be used in other contexts. Having said that, it can receive just an `address` parameter. 

Basically the name must not be related with the proyect domain.

Reply:
Ok i agree